Changes to survey-runner to use sdc-cryptography library

shahi645 opened the pull request:
### What is the context of this PR?
- removal of current eq cryptography code
- simplify code where possible
- update tests

### How to review 
Look through changes to code and run tests

jcox-dev commented on tests/integration/views/test_jwt_authentication.py:
```diff
@@ -28,6 +30,8 @@ def test_no_token(self):
 
     def test_invalid_token(self):
         token = "invalid"
+        # token = token.encode('utf-8')
```
Extra comment from dev left in.

jcox-dev commented on app/authentication/authenticator.py:
```diff
@@ -81,14 +82,14 @@ def decrypt_token(encrypted_token):
+    decrypted_token = JWTHelper.decode(jwt_token=jwt_token,
+                                       key_store=current_app.eq['secret_store'],
+                                       purpose=KEY_PURPOSE_AUTHENTICATION,
+                                       leeway=current_app.config['EQ_JWT_LEEWAY_IN_SECONDS'])
```
This repeated pattern of `decrypt` -> `decode` can be dropped in favour of the helper function from the lib, `sdc.crypto.decrypter.decrypt`.

jcox-dev commented on tests/integration/create_token.py:
```diff
@@ -63,6 +65,6 @@ def create_token_with_survey_url(self, form_type_id, eq_id, survey_url, **extra_
+        token = JWTHelper.encode(payload, self._upstream_kid, self._secret_store, KEY_PURPOSE_AUTHENTICATION)
 
-        return encrypt_jwe(token, self._sr_public_kid, self._secret_store, KEY_PURPOSE_AUTHENTICATION)
+        return JWEHelper.encrypt(token, self._sr_public_kid, self._secret_store, KEY_PURPOSE_AUTHENTICATION)
```
Likewise, `encode` -> `encrypt` can be dropped in favour of `sdc.crypto.encrypter.encrypt`.

jcox-dev reviewed (changes requested):
Change to use the helper methods `encrypt` and `decrypt` from the sdc-crypto lib and clean up extra comment. 

Attached suggested patch: [helper.txt](https://github.com/ONSdigital/eq-survey-runner/files/1333888/helper.txt).

ajmaddaford commented on app/views/session.py:
```diff
@@ -50,6 +51,8 @@ def login():
     except JtiTokenUsed as e:
         raise Unauthorized from e
+    except (TypeError, ValueError) as e:
+        raise InvalidTokenException
```
This should be `raise InvalidTokenException from e` to keep the traceback from the original exception.

jonnyshaw89 commented on app/secrets.py:
```diff
@@ -1,6 +1,6 @@
 from jwcrypto import jwk
```
All of this logic has also been replaced in the new library. or are we blocked from doing this until the secrets and keys have been split?

shahi645 replied:
yeah this is blocked until we split the keys from secrets. Card in ready for dev currently

jonnyshaw89 commented on tests/app/authentication/test_invalid_token_exception.py:
```diff
@@ -1,6 +1,6 @@
 import unittest
```
This test now seems pointless?we are testing an exception that is provided by the library.

shahi645 replied:
removed

shahi645 committed:
Changes to survey-runner to use sdc-cryptography library

    which include:

    - removal of current eq cryptography code
    - simplify code where possible
    - updated tests and requirements.txt

 #   removed test thats not needed
